tests/testthat/test-indentation_linter.R

```diff
@@ -0,0 +1,423 @@
+test_that("indentation linter flags unindented expressions", {
+  linter <- indentation_linter(indent = 2L)
+
+  expect_lint(
+    trim_some("
+      lapply(1:10, function(i) {
+        i %% 2
+      })
+    "),
+    NULL,
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+      lapply(1:10, function(i) {
+       i %% 2  # indentation is only 1 character
+      })
+    "),
+    "Indentation",
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+      lapply(1:10, function(i) {
+          i %% 2
+      })
+    "),
+    NULL,
+    indentation_linter(indent = 4L)
+  )
+
+  expect_lint(
+    trim_some("
+      lapply(1:10, function(i) {
+        i %% 2  # indentation is only 2 characters
+      })
+    "),
+    "Indentation",
+    indentation_linter(indent = 4L)
+  )
+
+  # ugly code, but still correctly indented
+  expect_lint(
+    trim_some("
+      list(
+           1,
+           2)
+    "),
+    NULL,
+    linter
+  )
+
+  # comments do not trigger hanging indent rule
+  expect_lint(
+    trim_some("
+      list( # comment
+        ok
+      )
+    "),
+    NULL,
+    linter
+  )
+
+  # assignment triggers indent
+  expect_lint(
+    trim_some("
+      a <-
+        expr(
+          42
+        )
+    "),
+    NULL,
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+      if (cond)
+        code
+
+      if (cond) code else code2
+
+      if (cond) {
+        code
+      } else
+        code
+
+      if (cond) {
+        code
+      } else {
+        code
+      }
+    "),
+    NULL,
+    linter
+  )
+})
+
+test_that("indentation linter flags improper closing curly braces", {
+  linter <- indentation_linter(indent = 2L)
+  expect_lint(
+    trim_some("
+      lapply(1:10, function(i) {
+        {
+          i %% 2
+        }
+      })
+    "),
+    NULL,
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+      lapply(1:10, function(i) {
+        i %% 2
+        } # closing curly doesn't return to parent indentation
+      )
+    "),
+    "Indentation",
+    linter
+  )
+})
+
+test_that("function argument indentation works in tidyverse-style", {
+  linter <- indentation_linter()
+  expect_lint(
+    trim_some("
+      function(a = 1L,
+               b = 2L) {
+        a + b
+      }
+    "),
+    NULL,
+    linter
+  )
+
+  expect_lint(
```

Comment:
this is just the previous example, plus assignment, right? could you add a quick comment to the test?